core/src/main/java/org/infinispan/stream/impl/local/EntryStreamSupplier.java

```diff
@@ -44,7 +45,9 @@ public EntryStreamSupplier(Cache<K, V> cache, ConsistentHash hash, Supplier<Stre
       }
       if (segmentsToFilter != null && hash != null) {
          log.tracef("Applying segment filter %s", segmentsToFilter);
-         stream = stream.filter(k -> segmentsToFilter.contains(hash.getSegment(k.getKey())));
+         BitSet bitSet = new BitSet(hash.getNumSegments());
```

Comment:
I will think it over a bit more, this seemed like the easiest place since it is also done lazily.  Unless you are saying so the BitSet creation is only done in one place?

Also I am not sure what toString you are talking about.  The tracef doesnt' do a toString unless trace is enabled.  However I can a trace variable that will avoid the logging level lookup.